lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -679,16 +679,22 @@ def pending_result(response, action)
       end
 
       def headers(action, options)
-        auth_token = @options[:access_token] ? "Bearer #{@options[:access_token]}" : @options[:secret_key]
-        auth_token = @options[:public_key] if action == :tokens
-        headers = {
-          'Authorization' => auth_token,
-          'Content-Type' => 'application/json;charset=UTF-8'
-        }
+        headers = { 'Authorization' => auth_token(action), 'Content-Type' => 'application/json;charset=UTF-8' }
         headers['Cko-Idempotency-Key'] = options[:idempotency_key] if options[:idempotency_key]
+
         headers
       end
 
+      def auth_token(action)
```

Comment:
just one small request, can you add some unit tests that verify we are passing Bearer in Authorization

Reply:
[done](https://github.com/activemerchant/active_merchant/pull/5381/files#diff-21853699189e55fa3c1e3c86bf42f1f09a2cb5766fe443284951350cab4f2420R700)!